packages/app/src/components/LinkTiles/LinkTile.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { EntityLink } from '@backstage/catalog-model';
+import { Link } from '@backstage/core-components';
+import { useApp } from '@backstage/core-plugin-api';
+import { BackstageTheme } from '@backstage/theme';
+import { Box, createStyles, makeStyles, Typography } from '@material-ui/core';
+import LanguageIcon from '@material-ui/icons/Language';
+import ArrowRightIcon from '@material-ui/icons/ArrowForwardIos';
+
+const useStyles = makeStyles((theme: BackstageTheme) =>
+  createStyles({
+    card: {
+      border: `1px solid ${theme.palette.divider}`,
+      boxShadow: theme.shadows[2],
+      borderRadius: '4px',
+      gap: '1em',
+      padding: theme.spacing(2),
+      color: '#fff',
+      transition: `${theme.transitions.duration.standard}ms`,
+      '&:hover': {
+        boxShadow: theme.shadows[4],
+      },
+      background: theme.page.backgroundImage,
+    },
+    link: {
+      '&:hover': {
+        textDecoration: 'none',
+      }
+    }
+  }),
+);
+
+const LinkTile = ({url, title, icon }: EntityLink) => {
+  const classes = useStyles();
+  const app = useApp();
+
+  const Icon = icon ? app.getSystemIcon(icon) ?? LanguageIcon : LanguageIcon;
```

Comment:
Can't we just use since it already has a default value with `??`?

```suggestion
  const Icon = app.getSystemIcon(icon) ?? LanguageIcon;
```

Reply:
As said in the PR description, this is exactly how it's defined in the `EntityLinksCard` in backstage upstream:

https://github.com/backstage/backstage/blob/2149f97b7301cb6b6ccf755d585d94ff715dc8ce/plugins/catalog/src/components/EntityLinksCard/EntityLinksCard.tsx#L38

Citing from the Upstream PR https://github.com/backstage/backstage/pull/4517

> Also updated app.getSystemIcon to accept undefined & return undefined. This avoids awkward typing on the consumer side as it will be common to have an optional icon & the possibility no system icon is defined. This forces the consumer to thinking about a fallback icon.

It all boils down to typing. `icon` parameter of `app.getSystemIcon` can't be `undefined`.

Reply:
Oh ok, thanks for the detailed explanation!